ini/parser.go

```diff
@@ -0,0 +1,317 @@
+package ini
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	s "strings"
+)
+
+// Helpers
+func contains(array []string, element string) bool {
+	for _, x := range array {
+		if x == element {
+			return true
+		}
+	}
+	return false
+}
+
+type Entry = map[string]string
+type INIMap = map[string]Entry
+
+// Parser
+type Parser struct {
+	parsedMap INIMap
+}
+
+/////////////////////
+// inner functions //
+/////////////////////
+
+func (p *Parser) addParent(parent string) {
+	// check if the parent does not exist in the parsed dict
+	if p.parsedMap[parent] == nil {
+		p.parsedMap[parent] = Entry{}
+	}
+}
+
+func (p *Parser) add(parent string, key string, value string) {
+	// add the parent first
+	p.addParent(parent)
+
+	p.parsedMap[parent][key] = value
+}
+
+////////////////////
+// user functions //
+////////////////////
+
+func (p *Parser) GetParsedMap() INIMap {
+	return p.parsedMap
+}
+
+func (p *Parser) ToString() string {
+	var parsedStr string = ""
```

Comment:
you can use the shorthand way `parsedStr:=""`

also it should be called `formattedStr`